client/trino-jdbc/src/main/java/io/trino/jdbc/AbstractTrinoResultSet.java

```diff
@@ -380,6 +372,23 @@ private static Date parseDate(String value, DateTimeZone localTimeZone)
         return new Date(calendar.getTimeInMillis());
     }
 
+    private static Optional<Date> tryParseDate(String value, DateTimeZone localTimeZone)
+    {
+        try {
+            long millis = DATE_FORMATTER.withZone(localTimeZone).parseMillis(String.valueOf(value));
+            if (millis >= START_OF_MODERN_ERA_SECONDS * MILLISECONDS_PER_SECOND) {
+                return Optional.of(new Date(millis));
+            }
+        }
+        catch (IllegalArgumentException ignored) {
+            // https://joda-time.sourceforge.net/faq.html#illegalinstant
+            LocalDate localDate = DATE_FORMATTER.parseLocalDate(String.valueOf(value));
```

Comment:
Why not to use this always?

Reply:
It won't work for every date (pre-gregorian ones)